src/__tests__/to-be-partially-checked.js

```diff
@@ -0,0 +1,124 @@
+import {render} from './helpers/test-utils'
+
+describe('.toBePartiallyChecked', () => {
+  test('handles input checkbox with aria-checked', () => {
+    const {queryByTestId} = render(`
+      <input type="checkbox" aria-checked="mixed" data-testid="checkbox-mixed" />
```

Comment:
Hmmm, I don't think we should support `aria-checked` in a `<input type="checkbox" />`. It should be supported only on non-native checkbox elements with `role="checkbox"`.

The reason being that a `<input type="checkbox" />` won't convey to the user visually that it is partially checked. The only way the native checkbox can be partially checked is when its  `indeterminate` property is set to `true`.

What do you think?

Reply:
I thought the same when I tested it in the broswer, as you say `indeterminate` set to `true` has a visual effect on the checkbox:

![image](https://user-images.githubusercontent.com/905225/81648960-d704a100-942f-11ea-9126-7ba274a8952c.png)

The only thing that made me change my mind is that in the accessibility consoleboth of them had role checkbox and checked `mixed`, but yeah, I think it should be better to check for input+indeterminate and role checkbox + aria mixed, let me change it

Reply:
> in the accessibility console both of them had role checkbox and checked mixed

Would you care to share more about this detail?

Reply:
Both inputs show the same data in the _Accessibility_ tab in the Chrome Devtools:

Input with `indeterminate`:

![image](https://user-images.githubusercontent.com/905225/81699693-f9b2ac00-9467-11ea-95b1-0a3fd38b60e2.png)

Input with `aria-checked="mixed"`:

![image](https://user-images.githubusercontent.com/905225/81699747-07683180-9468-11ea-959d-5e8a5c435ab6.png)

You can check the [Codesandbox demo](https://codesandbox.io/s/hungry-night-iwlkp?file=/src/App.js) I've been using.

Reply:
Ok, my bad then. If the `<input type="checkbox" aria-checked="mixed" />` shows as `checked: mixed` even if `indeterminate` was not set true, then I think it was ok the first time around, and I assumed without checking what you correctly went out to check.

What do you think?

If you agree, can you easily revert the changes after my earlier review? Sorry for having made you work extra for nothing.

Reply:
So we trust the accessibility info instead of how it's displayed right?

Reply:
Yes, I was also asking for your opinion here. But I'm conflicted. The `<input type="checkbox" aria-checked="mixed" />` looks visually as if unchecked, but semantically (from the aria perspective) it is still partially checked.

I'm leaning towards trusting the accessibility info. If someone uses an input checkbox with `aria-checked="mixed"` they would presumably take over the rendering to reflect that using icons and stuff.

So yes, in the end I'm leaning to that (after having thought about it out loud here 😄)

Reply:
Yeah, I thought the same at first, since it looks unchecked, but then I thought it's just a matter of how chrome styles the checkbox.

Let's support both of them then :)

Reply:
Ok, so the last commit needs to be reverted, right? It was good the first time around, right?

Reply:
Yes, I'll revert it and change only the check inside `isValidAriaElement`

Reply:
Should be good now :)